Dockerfile

```diff
@@ -1,26 +1,81 @@
 # https://tracker.mender.io/browse/MEN-2399
-RUN wget -nc -q https://toolchains.bootlin.com/downloads/releases/toolchains/armv6-eabihf/tarballs/armv6-eabihf--glibc--stable-2018.11-1.tar.bz2 \
-    && tar -xjf armv6-eabihf--glibc--stable-2018.11-1.tar.bz2 \
-    && rm armv6-eabihf--glibc--stable-2018.11-1.tar.bz2
+ENV ARMV6_TOOLCHAIN_ROOT="/armv6-eabihf--glibc--stable-2018.11-1"
+RUN if [ "${ARCH}" = "armhf" ]; then \
+        wget -nc -q https://toolchains.bootlin.com/downloads/releases/toolchains/armv6-eabihf/tarballs/armv6-eabihf--glibc--stable-2018.11-1.tar.bz2 \
+        && tar -xjf armv6-eabihf--glibc--stable-2018.11-1.tar.bz2 \
+        && rm armv6-eabihf--glibc--stable-2018.11-1.tar.bz2; \
+    fi
+
+# Get depdendencies from upstream, manually donwloading deb packages, and fake pkg-config.
+# NOTE: pkg-config is used from go build to obtain cflags and libs required to build C code; however
+# for armhf we will explicitly pass these flags for go to use our custom sysroot instead of the system
+# one. Alternatively pkg-config supports setting a prefix, but that would require patching the source code.
+RUN if [ "${ARCH}" = "armhf" ]; then \
+        set -e; \
+        ln -s /bin/true /usr/bin/pkg-config; \
+        curl -f http://raspbian.raspberrypi.org/raspbian/dists/stretch/main/binary-armhf/Packages -o Packages 2>/dev/null; \
+        for pkg in \
+            liblzma5 \
+            liblzma-dev \
+            libssl1.1 \
+            libssl-dev \
+            libglib2.0-0 \
+            libglib2.0-dev \
+            libpcre3 \
+            libpcre3-dev \
+            zlib1g \
+            zlib1g-dev \
+            libffi6 \
+            libffi-dev \
+            libmount1 \
+            libmount-dev \
+            libselinux1 \
+            libselinux1-dev \
+            libblkid1 \
+            libblkid-dev \
+            libuuid1 \
+            uuid-runtime \
+            uuid-dev \
+        ; do \
+            deb_package_url=$(grep Filename Packages | grep ${pkg}_ | grep armhf | tail -n1 | sed 's/Filename: //'); \
+            echo "Downloading ${pkg}..."; \
+            filename=$(basename $deb_package_url); \
+            curl -L http://raspbian.raspberrypi.org/raspbian/${deb_package_url} -o $filename 2>/dev/null; \
+            echo "Extracting ${pkg}..."; \
+            ar -x ${pkg}_*_armhf.deb data.tar.xz; \
+            for dir in ./lib ./usr/include ./usr/lib; do \
+                ( tar -tf data.tar.xz ${dir} >/dev/null 2>/dev/null ) && \
+                ( echo "   ${dir}..."; \
+                  tar -C ${ARMV6_TOOLCHAIN_ROOT} \
+                    -xf data.tar.xz  ${dir} ); \
+            done; \
+        done; \
+        rm -f Packages data.tar.xz *.deb; \
+    fi
+
+RUN if [ "${ARCH}" = "arm64" ]; then \
+        apt-get install -y gcc-aarch64-linux-gnu; \
+    fi
 
 # Golang environment, for cross-compiling the Mender client
 ARG GOLANG_VERSION=1.11.5
```

Comment:
Question:
Is it on purpose that we're still on this old an version of Go, or have we just not updated in a while?

Reply:
Not on purpose, just something that we have not updated in a long while. I will follow-up in a separate PR for this, aligning it with today's yocto `dunfell` golang's version.